arcade/physics_engines.py

```diff
@@ -53,7 +54,7 @@ def _circular_check(player: Sprite, walls: List[SpriteList]):
         vary *= 2
 
 
-def _move_sprite(moving_sprite: Sprite, walls: List[SpriteList], ramp_up: bool) -> List[Sprite]:
+def _move_sprite(moving_sprite: Sprite, walls: List[SpriteList], ramp_up: bool) -> List[Union[Sprite, BasicSprite]]:
```

Comment:
```suggestion
def _move_sprite(moving_sprite: Sprite, walls: List[SpriteList[SpriteType]], ramp_up: bool) -> List[SpriteType]:
```

I think this should be generic.  The returned list contains wall sprites, so it'll return whatever type of BasicSprite (sub)class is contained in the `walls` SpriteLists.

Reply:
@DragonMoffon Bumping this since this PR it would fix issues which flag #1876 as broken.